src/validation.h

```diff
@@ -744,6 +745,62 @@ bool InvalidateBlock(CValidationState& state, const CChainParams& chainparams, C
 /** Remove invalidity status from a block and its descendants. */
 void ResetBlockFailureFlags(CBlockIndex* pindex) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+/**
+ * Serialized version of a UTXO set from which an assumed-valid CChainState
+ * can be constructed.
+ */
+class SnapshotMetadata
```

Comment:
Could also create `src/node/utxo_snapshot.{h,cpp}` and add this there if that seems preferable.